### DISCUSSION TITLE
Get upload's full data in upload field's cellData prop

### DISCUSSION THREAD

**Discussion by qbeauperin** [payloadcms/payload#2389] [Feature Requests & Ideas]
Right now, the [`cellData` prop of the Cell](https://payloadcms.com/docs/admin/components#cell-component) of an [Upload field](https://payloadcms.com/docs/fields/upload) only returns the doc id.

Thoughts on adding some `depth` to whatever the query that's getting that data is in order to get the full upload's data?

Use case example: displaying the thumbnail of a post in the list view.

**Comment by DanRibbens** [comment]
Hey @qbeauperin,
Cell data for upload and relationship is fetched via the RelationshipProvider. That component is hard-coded to `depth: 0` for performance reasons and I don't know that changing it would be the best way here.

What I would do instead is add an additional field for the thumbnail on post that is set from an `afterChange` hook on the posts collection. That way you would have the thumbnail without needing to populate more from the database.

What do you think about that approach?

**Reply by qbeauperin** [reply]
> That component is hard-coded to depth: 0 for performance reasons and I don't know that changing it would be the best way here.

Ah I see, that's fair! 👍 

> What do you think about that approach?

As a general solve for this post's specific question, I think this could definitely work!

Although for the "thumbnail display in list view" use case, I'm assuming that there's probably quite a few people that would expect this to be a default behavior of the upload field's cell component, and having to do this whole setup for every upload field you have might not be the best DX. 🤔 

I've already made a feature request for this specific use case though - #2393 - so I think this one can be closed!
